### ISSUE TITLE
Add tests

### ISSUE THREAD

**Issue by gaearon** [gaearon/react-hot-api#1]
It would be much easier now due to cleaner separation.

Here's what tests should do:

* Have “versions” of classes, like `ComponentBeforeUpdate`, `ComponentAfterUpdate`
* Test that certain behavior changes after applying update:
  - `render` returns a different tree;
  - methods previously put into `setTimeout` also execute new code after update.
* Verify that component doesn't lose state and doesn't unmount

Basically verify that patching works correctly. Edge cases to consider:

* All component styles (`createClass`, `extends React.Component` in 0.13+, plain object in 0.13+)
* Components with `shouldComponentUpdate` returning `false`

**Comment by gaearon** [comment]
The implementation will continue in the spinoff project: [react-hotify](https://github.com/gaearon/react-hotify). When it matures, it will replace React Hot API.
